Compiler error: EnrollmentAction and DeleteAction::play signature mismatch with Action::play in hlk_fm22x component

### The problem

When compiling a project that uses the HLK-FM22x component, the implementations of play(...) in EnrollmentAction and DeleteAction do not match the signature of the base class Action<Ts...>. The base declaration requires:

    virtual void play(const Ts &...x) = 0;

The component currently implements:

    void play(Ts... x) override { ... }

As a result, the derived classes remain abstract and cannot be instantiated. Compiler errors occur such as:

src/esphome/components/hlk_fm22x/hlk_fm22x.h:192:8: error: 'void esphome::hlk_fm22x::EnrollmentAction<Ts>::play(Ts ...) [...]' marked 'override', but does not override
src/esphome/components/hlk_fm22x/hlk_fm22x.h:203:8: error: 'void esphome::hlk_fm22x::DeleteAction<Ts>::play(Ts ...) [with Ts = {long int}]' marked 'override', but does not override

Expected behavior:
The play methods in hlk_fm22x should be declared as 'void play(const Ts& ... x) override' so the concrete classes compile correctly.

### Which version of ESPHome has the issue?

2025.11.2

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP32

### Component causing the issue

hlk_fm22x

### YAML Config

```yaml
substitutions:
  devicename: frontdoor
  friendly_name: "Haustüre Einlass"

esphome:
  name: ${devicename}

esp32:
  board: esp32dev
  framework:
    type: esp-idf      

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  ap:
    ssid: ${devicename} Hotspot
    password: !secret wifi_password

captive_portal:

logger:
  level: DEBUG
  baud_rate: 0        # keine serielle Ausgabe → UART0 frei

ota:
  password: !secret esp_password
  platform: esphome

api:
  encryption:
    key: !secret api_key

  actions:
  - action: enroll
    variables:
      name: string
      direction: int
    then:
      - hlk_fm22x.enroll:
          name: !lambda 'return name;'
          direction: !lambda 'return direction;'
  - action: scan
    then:
      - hlk_fm22x.scan:
  - action: delete
    variables:
      face_id: int
    then:
      - hlk_fm22x.delete:
          face_id: !lambda 'return face_id;'
  - action: delete_all
    then:
      - hlk_fm22x.delete_all:
  - action: reset
    then:
      - hlk_fm22x.reset:


#BLE-Tracker-Hub (nutzt das interne BT)
#   esp32_ble_tracker:

uart:
  rx_pin: GPIO18
  tx_pin: GPIO19
  baud_rate: 115200

hlk_fm22x:
  on_face_scan_invalid:
    - homeassistant.event:
        event: esphome.test_node_face_scan_invalid
        data:
          error: !lambda 'return error;'
  on_face_scan_matched:
    - homeassistant.event:
        event: esphome.test_node_face_scan_matched
        data:
          face_id: !lambda 'return face_id;'
          name: !lambda 'return name;'
  on_face_scan_unmatched:
    - homeassistant.event:
        event: esphome.test_node_face_scan_unmatched
  on_face_info:
    - homeassistant.event:
        event: esphome.test_node_face_info
        data:
          status: !lambda 'return status;'
          left: !lambda 'return left;'
          top: !lambda 'return top;'
          right: !lambda 'return right;'
          bottom: !lambda 'return bottom;'
          yaw: !lambda 'return yaw;'
          pitch: !lambda 'return pitch;'
          roll: !lambda 'return roll;'
  on_enrollment_done:
    - homeassistant.event:
        event: esphome.test_node_enrollment_done
        data:
          face_id: !lambda 'return face_id;'
          direction: !lambda 'return direction;'
  on_enrollment_failed:
    - homeassistant.event:
        event: esphome.test_node_enrollment_failed
        data:
          error: !lambda 'return error;'
```

### Anything in the logs that might be useful for us?

```txt
INFO ESPHome 2025.11.2
INFO Reading configuration /config/esphome/frontdoor.yaml...
INFO Generating C++ source...
INFO Setting CONFIG_LWIP_MAX_SOCKETS to 11 (registered: api=4, captive_portal=4, mdns=2, ota=1)
INFO Compiling app... Build path: /data/build/frontdoor
Processing frontdoor (board: esp32dev; framework: espidf; platform: https://github.com/pioarduino/platform-espressif32/releases/download/55.03.31-2/platform-espressif32.zip)
--------------------------------------------------------------------------------
INFO Package configuration completed successfully
INFO Package configuration completed successfully
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - contrib-piohome @ 3.4.4 
 - framework-espidf @ 3.50501.0 (5.5.1) 
 - tool-cmake @ 4.0.3 
 - tool-esp-rom-elfs @ 2024.10.11 
 - tool-esptoolpy @ 5.1.0 
 - tool-mklittlefs @ 3.2.0 
 - tool-ninja @ 1.13.1 
 - tool-scons @ 4.40801.0 (4.8.1) 
 - toolchain-xtensa-esp-elf @ 14.2.0+20241119
Reading CMake configuration...
Dependency Graph
|-- noise-c @ 0.1.10
Compiling .pioenvs/frontdoor/src/main.cpp.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_end.c.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_file.c.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_hashtable.c.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_int.c.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_lit.c.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_rem.c.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_rex.c.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_str.c.o
Compiling .pioenvs/frontdoor/console/argtable3/arg_utils.c.o
Compiling .pioenvs/frontdoor/console/argtable3/argtable3.c.o
Compiling .pioenvs/frontdoor/cxx/cxx_exception_stubs.cpp.o
Compiling .pioenvs/frontdoor/cxx/cxx_guards.cpp.o
Compiling .pioenvs/frontdoor/cxx/cxx_init.cpp.o
Compiling .pioenvs/frontdoor/driver/deprecated/adc_legacy.c.o
Archiving .pioenvs/frontdoor/esp-idf/console/libconsole.a
Compiling .pioenvs/frontdoor/driver/deprecated/adc_dma_legacy.c.o
Archiving .pioenvs/frontdoor/esp-idf/cxx/libcxx.a
Compiling .pioenvs/frontdoor/driver/deprecated/dac_common_legacy.c.o
Compiling .pioenvs/frontdoor/driver/deprecated/esp32/dac_legacy.c.o
Compiling .pioenvs/frontdoor/driver/deprecated/timer_legacy.c.o
Compiling .pioenvs/frontdoor/driver/i2c/i2c.c.o
Compiling .pioenvs/frontdoor/driver/deprecated/i2s_legacy.c.o
Compiling .pioenvs/frontdoor/driver/deprecated/mcpwm_legacy.c.o
In file included from src/esphome.h:27,
                 from src/main.cpp:3:
src/esphome/components/hlk_fm22x/hlk_fm22x.h: In instantiation of 'class esphome::hlk_fm22x::EnrollmentAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long int>':
src/main.cpp:270:89:   required from here
  270 |   hlk_fm22x_enrollmentaction_id = new hlk_fm22x::EnrollmentAction<std::string, int32_t>();
      |                                                                                         ^
src/esphome/components/hlk_fm22x/hlk_fm22x.h:192:8: error: 'void esphome::hlk_fm22x::EnrollmentAction<Ts>::play(Ts ...) [with Ts = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long int}]' marked 'override', but does not override
  192 |   void play(Ts... x) override {
      |        ^~~~
src/main.cpp: In function 'void setup()':
src/main.cpp:270:89: error: invalid new-expression of abstract class type 'esphome::hlk_fm22x::EnrollmentAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long int>'
  270 |   hlk_fm22x_enrollmentaction_id = new hlk_fm22x::EnrollmentAction<std::string, int32_t>();
      |                                                                                         ^
src/esphome/components/hlk_fm22x/hlk_fm22x.h:187:32: note:   because the following virtual functions are pure within 'esphome::hlk_fm22x::EnrollmentAction<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long int>':
  187 | template<typename... Ts> class EnrollmentAction : public Action<Ts...>, public Parented<HlkFm22xComponent> {
      |                                ^~~~~~~~~~~~~~~~
In file included from src/esphome/components/api/api_server.h:9,
                 from src/esphome/components/api/api_connection.h:8,
                 from src/esphome.h:3:
src/esphome/core/automation.h:225:16: note:     'void esphome::Action<Ts>::play(const Ts& ...) [with Ts = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long int}]'
  225 |   virtual void play(const Ts &...x) = 0;
      |                ^~~~
src/esphome/components/hlk_fm22x/hlk_fm22x.h: In instantiation of 'class esphome::hlk_fm22x::DeleteAction<long int>':
/config/esphome/frontdoor.yaml:53:68:   required from here
   53 |   - action: delete_all
      |                                                                    ^
src/esphome/components/hlk_fm22x/hlk_fm22x.h:203:8: error: 'void esphome::hlk_fm22x::DeleteAction<Ts>::play(Ts ...) [with Ts = {long int}]' marked 'override', but does not override
  203 |   void play(Ts... x) override {
      |        ^~~~
/config/esphome/frontdoor.yaml:53:68: error: invalid new-expression of abstract class type 'esphome::hlk_fm22x::DeleteAction<long int>'
   53 |   - action: delete_all
      |                                                                    ^
src/esphome/components/hlk_fm22x/hlk_fm22x.h:199:32: note:   because the following virtual functions are pure within 'esphome::hlk_fm22x::DeleteAction<long int>':
  199 | template<typename... Ts> class DeleteAction : public Action<Ts...>, public Parented<HlkFm22xComponent> {
      |                                ^~~~~~~~~~~~
src/esphome/core/automation.h:225:16: note:     'void esphome::Action<Ts>::play(const Ts& ...) [with Ts = {long int}]'
  225 |   virtual void play(const Ts &...x) = 0;
      |                ^~~~
Compiling .pioenvs/frontdoor/driver/deprecated/pcnt_legacy.c.o
Compiling .pioenvs/frontdoor/driver/deprecated/rmt_legacy.c.o
Compiling .pioenvs/frontdoor/driver/deprecated/sigma_delta_legacy.c.o
Compiling .pioenvs/frontdoor/driver/touch_sensor/touch_sensor_common.c.o
Compiling .pioenvs/frontdoor/driver/touch_sensor/esp32/touch_sensor.c.o
*** [.pioenvs/frontdoor/src/main.cpp.o] Error 1
```

### Additional information

See suggested change:

In src/esphome/components/hlk_fm22x/hlk_fm22x.h the methods

    void play(Ts... x) override { ... }

should be changed to

    void play(const Ts &... x) override { ... }

--- a/esphome/components/hlk_fm22x/hlk_fm22x.h
+++ b/esphome/components/hlk_fm22x/hlk_fm22x.h
@@ -189,7 +189,7 @@ template<typename... Ts> class EnrollmentAction : public Action<Ts...>, public P
   TEMPLATABLE_VALUE(std::string, name)
   TEMPLATABLE_VALUE(uint8_t, direction)
 
-  void play(Ts... x) override {
+  void play(const Ts &...x) override {
     auto name = this->name_.value(x...);
     auto direction = (HlkFm22xFaceDirection) this->direction_.value(x...);
     this->parent_->enroll_face(name, direction);
@@ -200,7 +200,7 @@ template<typename... Ts> class DeleteAction : public Action<Ts...>, public Paren
  public:
   TEMPLATABLE_VALUE(int16_t, face_id)
 
-  void play(Ts... x) override {
+  void play(const Ts &...x) override {
     auto face_id = this->face_id_.value(x...);
     this->parent_->delete_face(face_id);
   }
@@ -208,17 +208,17 @@ template<typename... Ts> class DeleteAction : public Action<Ts...>, public Paren
 
 template<typename... Ts> class DeleteAllAction : public Action<Ts...>, public Parented<HlkFm22xComponent> {
  public:
-  void play(Ts... x) override { this->parent_->delete_all_faces(); }
+  void play(const Ts &...x) override { this->parent_->delete_all_faces(); }
 };
 
 template<typename... Ts> class ScanAction : public Action<Ts...>, public Parented<HlkFm22xComponent> {
  public:
-  void play(Ts... x) override { this->parent_->scan_face(); }
+  void play(const Ts &...x) override { this->parent_->scan_face(); }
 };
 
 template<typename... Ts> class ResetAction : public Action<Ts...>, public Parented<HlkFm22xComponent> {
  public:
-  void play(Ts... x) override { this->parent_->reset(); }
+  void play(const Ts &...x) override { this->parent_->reset(); }
 };
 
 }  // namespace esphome::hlk_fm22x

